### PR TITLE
Commit vote for a ballot

### PR DESCRIPTION
**Pull request by kangarang**
- add voting logic
- get votingRights in eth context
- connect token contract to signer
- change slate.id -> actual slateID
- increase specificity of ethers function calls
- add metadataHash field to slates
- add /abis, import from /abis
- update abis
- add _docs folder w/ epoch lifecycle information
- rm development from migrations devNetworks
- add favicon

### CODE REVIEW

File: api/utils/slates.js [hunk 1]
```diff
@@ -22,13 +22,13 @@ async function getAllSlates() {
   const gatekeeper = new ethers.Contract(gatekeeperAddress, Gatekeeper.abi, provider);
 
   // Get an interface to the ParameterStore contract
-  const parameters = await gatekeeper.parameters();
-  const parameterStore = new ethers.Contract(parameters, ParameterStore.abi, provider);
+  const parameterStoreAddress = await gatekeeper.functions.parameters();
+  const parameterStore = new ethers.Contract(parameterStoreAddress, ParameterStore.abi, provider);
   // Get the slate staking requirement
-  const requiredStake = await parameterStore.get('slateStakeAmount');
+  const requiredStake = await parameterStore.functions.get('slateStakeAmount');
```
Comment (akuanti) [inline api/utils/slates.js]: Is there a difference between using `parameterStore.get()` and `parameterStore.functions.get()`?

File: client/pages/ballots/vote.tsx [hunk 2]
```diff
@@ -111,37 +113,41 @@ const Vote: React.FunctionComponent<IProps> = ({ router }) => {
...
+      if (votingRights.gt('0') && panBalance.eq('0')) {
+        console.log('only votingRights');
+        // entire balance is being used as votingRights
+        // -> vote w/ votingRights
+        numTokens = votingRights;
+      } else if (votingRights.gt('0') && panBalance.gt('0')) {
+        console.log('both votingRights and user balance');
+        // balance is split between gate_keeper and user_account
+        // -> deposit more tokens / get more votingRights, then vote with entire (new) votingRights
+        await contracts.gateKeeper.functions.depositVoteTokens(panBalance);
```
Comment (akuanti) [inline client/pages/ballots/vote.tsx]: There isn't any need to explicitly deposit more tokens, as that automatically happens (assuming the Gatekeeper's allowance is high enough).

File: client/pages/ballots/vote.tsx [hunk 3]
```diff
@@ -111,37 +113,41 @@ const Vote: React.FunctionComponent<IProps> = ({ router }) => {
...
-
-      // console.log('numTokens:', numTokens);
-
-      // if (numTokens.gt('0')) {
-      //   // commit the ballot to the gateKeeper contract
-      //   await contracts.gateKeeper.functions.commitBallot(commitHash, numTokens);
-      // }
+      let numTokens: utils.BigNumber = utils.bigNumberify('0');
+
+      // NOTE: this userflow might need revision
```
Comment (akuanti) [inline client/pages/ballots/vote.tsx]: The main revision I can imagine is that the user might want to specify how many tokens to use for voting, but we can add that later.

File: client/utils/provider.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import getConfig from 'next/config';
 
 const TokenCapacitor = require('./abis/TokenCapacitor.json');
 const Gatekeeper = require('./abis/Gatekeeper.json');
-// const Token = require('./abis/IERC20.json');
+const Token = require('./abis/BasicToken.json');
```
Comment (akuanti) [inline client/utils/provider.ts]: I think it might be better to use `IERC20` here, as it's more generic.

File: api/utils/slates.js (hunk 1 above)
Comment (kangarang) [reply api/utils/slates.js]: no difference. adding `functions` is just more explicit

File: client/pages/ballots/vote.tsx (hunk 2 above)
Comment (kangarang) [reply client/pages/ballots/vote.tsx]: i see. should `numTokens = panBalance.add(votingRights)` then?

File: client/utils/provider.ts (hunk 4 above)
Comment (kangarang) [reply client/utils/provider.ts]: when the json files are built during a migration, the `networks` field of `BasicToken` is filled in, while the `networks` field of `IERC20` is not. that value could potentially be useful. what do you think?

**Pull request merged**